lib/frameworks/utils.rb

```diff
@@ -95,5 +95,34 @@ def save_and_link_screenshot
       Capybara.current_session.driver.save_screenshot("./reports/Screenshot_#{current_time}.png")
       embed "./reports/Screenshot_#{current_time}.png", 'image/png', "Actual screenshot of the error at #{current_url}"
     end
+
+    def getting_console_errors(type)
+    case type
+    when 'errors'
+      errors = page.driver.browser.manage.logs.get(:browser).select {|e| e.level == "SEVERE"}.map(&:message).to_a
+       if errors.present?
+         puts "Below are the errors found."
+         raise StandardError, errors.join("\n\n")
+       else
+         puts "No Error found in console"
+       end
+     when 'warnings'
+       warnings = page.driver.browser.manage.logs.get(:browser).select {|e| e.level != "SEVERE"}.map(&:message).to_a
+       if warnings.present?
+         puts "Below are the warnings found."
+         raise StandardError, warnings.join("\n\n")
+       else
+         puts "No warnings found in console"
+       end
+     when 'all'
```

Comment:
'all' is not appropriate here for 'type'. Lets discuss on this

Reply:
agree, we can say something 'errors_and_warnings'